src/components/button/button.component.tsx

```diff
@@ -237,6 +239,10 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
 
     const [internalRef, setInternalRef] = useState<HTMLButtonElement>();
 
+    const { inSplitButton, onChildButtonClick } = useContext(
```

Comment:
**comment**(non-blocking): I wonder if we need `inSplitButton` here at all. As `onChildButtonClick` is optional we could derive `inSplitButton` from it like so `const inSplitButton = !!onChildButtonClick;`.

Reply:
It's a fair comment. I like having the explicit boolean here rather than assume the presence of another property implies anything - as what if we later we want to make `onChildButtonClick` optional even in the SplitButton case. As ever I'm probably worrying about something that will never happen 😂 - but I don't see much harm in being explicit. Happy to be persuaded otherwise though.

Reply:
Happy to let the second reviewer split the difference 😄

Reply:
Personally, I prefer `const inSplitButton = !!onChildButtonClick;` approach for this particular need. As we don't have any future plans plans to change `onChildButtonClick` and if we did, we could always implement what you have here but seeing as @edleeks87 has already approved the PR, I'm not going to block the PR/request changes over it. Just my preference and not a hill I want to die on 😆

Reply:
Personally, I prefer `const inSplitButton = !!onChildButtonClick;` approach for this particular need. As we don't have any future plans plans to change `onChildButtonClick` and if we did, we could always implement what you have here but seeing as @edleeks87 has already approved the PR, I'm not going to block the PR/request changes over it. Just my preference and not a hill I want to die on 😆